Content.Server/Bed/BedSystem.cs

```diff
@@ -0,0 +1,117 @@
+using Content.Shared.Damage;
+using Content.Server.Bed.Components;
+using Content.Server.Buckle.Components;
+using Content.Server.Body.Systems;
+using Content.Shared.Buckle.Components;
+using Content.Shared.Body.Components;
+using Content.Shared.Bed;
+using Content.Server.Power.Components;
+using Content.Shared.Emag.Systems;
+
+namespace Content.Server.Bed
+{
+    public sealed class BedSystem : EntitySystem
+    {
+        [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<StasisBedComponent, ComponentStartup>(OnComponentStartup);
+            SubscribeLocalEvent<StasisBedComponent, BuckleChangeEvent>(OnBuckleChange);
+            SubscribeLocalEvent<StasisBedComponent, PowerChangedEvent>(OnPowerChanged);
+            SubscribeLocalEvent<StasisBedComponent, GotEmaggedEvent>(OnEmagged);
+        }
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            foreach (var (bedComponent, strapComponent) in EntityManager.EntityQuery<HealOnBuckleComponent, StrapComponent>())
+            {
+                if (strapComponent.BuckledEntities.Count == 0)
+                {
+                    bedComponent.Accumulator = 0;
+                    continue;
+                }
+                bedComponent.Accumulator += frameTime;
+
+                if (bedComponent.Accumulator < bedComponent.HealTime)
+                {
+                    continue;
+                }
+                bedComponent.Accumulator -= bedComponent.HealTime;
+                foreach (EntityUid healedEntity in strapComponent.BuckledEntities)
+                {
+                    _damageableSystem.TryChangeDamage(healedEntity, bedComponent.Damage, true);
+                }
+            }
```

Comment:
There's probably enough beds on a station that this shouldn't use EntityQuery<T> and should keep track of ones that are strapped.